tools/preflight-checks.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Will be set to false if any of the steps fail
+did_checks_pass=true
+
+# Minimum versions
+MIN_GO_VERSION="1.17"
+MIN_NODE_VERSION="14.0.0"
```

Comment:
This might be too much but it would be nice to parse these from a centralized place (e.g. package.json also has these values and the yarn install script)

Reply:
yeah we could do this in a follow up and require all scripts to use the versions defined in a central place